android/expoview/src/main/java/host/exp/expoview/Exponent.java

```diff
@@ -480,6 +483,25 @@ public void onCachedResponse(ExpoResponse response, boolean isEmbedded) {
     return sourceFile.exists();
   }
 
+  // As OTAs piles up, the JS bundles will consume quite an amount of storage space
+  // App developers, if find needed, can purge all the existing cache
+  public boolean clearAllJSBundleCache(final String abiVersion) throws IOException {
+    final File filesDir = mContext.getFilesDir();
+    final File directory = new File(filesDir, abiVersion);
+    if (!ABIVERSION_PATTERN.matcher(abiVersion).matches()) {
+      return false;
+    }
+    if (!directory.getCanonicalPath().startsWith(filesDir.getCanonicalPath())) {
+      return false;
+    }
+    if (directory.exists()) {
+      directory.delete();
```

Comment:
sorry, one more small change here 🙂 
```suggestion
      return directory.delete();
```
and then remove the following line.

Reply:
Gotcha.